### PR TITLE
Backport/dependant move

### PR DESCRIPTION
**Pull request by xavierh**
Depends of 1153
Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java [hunk 1]
```diff
@@ -1,6 +1,10 @@
 package com.minecolonies.coremod.entity.ai.citizen.builder;
 
+import com.minecolonies.api.util.BlockPosUtil;
 import com.minecolonies.api.util.BlockUtils;
+import com.minecolonies.api.util.EntityUtils;
+import com.minecolonies.api.util.InventoryUtils;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java]: MINOR Remove this unused import 'com.minecolonies.api.util.InventoryUtils'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/entity/pathfinding/WalkToProxy.java [hunk 2]
```diff
@@ -1,15 +1,17 @@
 package com.minecolonies.coremod.entity.pathfinding;
 
+import com.minecolonies.api.util.BlockPosUtil;
+import com.minecolonies.api.util.EntityUtils;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/pathfinding/WalkToProxy.java]: MINOR Remove this unused import 'com.minecolonies.api.util.EntityUtils'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingFarmer.java [hunk 3]
```diff
@@ -1,5 +1,8 @@
 package com.minecolonies.coremod.colony.buildings;
 
+import com.minecolonies.api.util.constants.ToolType;
+import com.minecolonies.api.util.ItemStackUtils;
+import com.minecolonies.api.util.Utils;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/BuildingFarmer.java]: MINOR Remove this unused import 'com.minecolonies.api.util.Utils'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/tileentities/ScarecrowTileEntity.java [hunk 4]
```diff
@@ -1,9 +1,10 @@
 package com.minecolonies.coremod.tileentities;
 
+import com.minecolonies.api.util.EntityUtils;
 import com.minecolonies.coremod.colony.Colony;
 import com.minecolonies.coremod.colony.ColonyManager;
 import com.minecolonies.coremod.inventory.InventoryField;
-import com.minecolonies.coremod.util.EntityUtils;
+import com.minecolonies.coremod.lib.Constants;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/tileentities/ScarecrowTileEntity.java]: MINOR Remove this unused import 'com.minecolonies.coremod.lib.Constants'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java [hunk 5]
```diff
@@ -1,13 +1,14 @@
 package com.minecolonies.coremod.tileentities;
 
+import com.minecolonies.api.util.constants.IToolType;
+import com.minecolonies.api.util.constants.ToolType;
+import com.minecolonies.api.util.InventoryFunctions;
+import com.minecolonies.api.util.ItemStackUtils;
+import com.minecolonies.api.util.InventoryUtils;
+import com.minecolonies.api.util.Utils;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java]: MINOR Remove this unused import 'com.minecolonies.api.util.Utils'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 209 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 121 major
* ![MINOR][MINOR] 63 minor
* ![INFO][INFO] 24 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [InventoryCitizen.java#L349](https://github.com/Minecolonies/minecolonies/blob/e1b63fa7684e02238fb12ade3ed8331f615c3f24/src/main/java/com/minecolonies/coremod/inventory/InventoryCitizen.java#L349): Reduce the number of conditional operators (7) used in the expression (maximum allowed 6). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1067)
1. ![MAJOR][MAJOR] [WindowHutFarmer.java#L206](https://github.com/Minecolonies/minecolonies/blob/e1b63fa7684e02238fb12ade3ed8331f615c3f24/src/main/java/com/minecolonies/coremod/client/gui/WindowHutFarmer.java#L206): Reduce this anonymous class number of lines from 37 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L859](https://github.com/Minecolonies/minecolonies/blob/e1b63fa7684e02238fb12ade3ed8331f615c3f24/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L859): Reduce this anonymous class number of lines from 24 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L956](https://github.com/Minecolonies/minecolonies/blob/e1b63fa7684e02238fb12ade3ed8331f615c3f24/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L956): Reduce this anonymous class number of lines from 40 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [CitizenData.java#L23](https://github.com/Minecolonies/minecolonies/blob/e1b63fa7684e02238fb12ade3ed8331f615c3f24/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L23): class "CitizenData" has 43 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/e1b63fa7684e02238fb12ade3ed8331f615c3f24/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.264 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L46](https://github.com/Minecolonies/minecolonies/blob/e1b63fa7684e02238fb12ade3ed8331f615c3f24/src/main/java/com/minecolonies/coremod/colony/Colony.java#L46): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 44 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L46](https://github.com/Minecolonies/minecolonies/blob/e1b63fa7684e02238fb12ade3ed8331f615c3f24/src/main/java/com/minecolonies/coremod/colony/Colony.java#L46): class "Colony" has 84 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L662](https://github.com/Minecolonies/minecolonies/blob/e1b63fa7684e02238fb12ade3ed8331f615c3f24/src/main/java/com/minecolonies/coremod/colony/Colony.java#L662): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L849](https://github.com/Minecolonies/minecolonies/blob/e1b63fa7684e02238fb12ade3ed8331f615c3f24/src/main/java/com/minecolonies/coremod/colony/Colony.java#L849): Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java [hunk 6]
```diff
@@ -1,13 +1,13 @@
 package com.minecolonies.coremod.tileentities;
 
+import com.minecolonies.api.util.constants.IToolType;
+import com.minecolonies.api.util.constants.ToolType;
+import com.minecolonies.api.util.InventoryFunctions;
+import com.minecolonies.api.util.ItemStackUtils;
+import com.minecolonies.api.util.InventoryUtils;
 import com.minecolonies.coremod.colony.Colony;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java]: MINOR Remove this unused import 'com.minecolonies.api.util.Utils'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 205 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 121 major
* ![MINOR][MINOR] 59 minor
* ![INFO][INFO] 24 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [InventoryCitizen.java#L349](https://github.com/Minecolonies/minecolonies/blob/301f1b31147e678aaa25b31c8e5d7d826ce1c233/src/main/java/com/minecolonies/coremod/inventory/InventoryCitizen.java#L349): Reduce the number of conditional operators (7) used in the expression (maximum allowed 6). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1067)
1. ![MAJOR][MAJOR] [WindowHutFarmer.java#L206](https://github.com/Minecolonies/minecolonies/blob/301f1b31147e678aaa25b31c8e5d7d826ce1c233/src/main/java/com/minecolonies/coremod/client/gui/WindowHutFarmer.java#L206): Reduce this anonymous class number of lines from 37 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L859](https://github.com/Minecolonies/minecolonies/blob/301f1b31147e678aaa25b31c8e5d7d826ce1c233/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L859): Reduce this anonymous class number of lines from 24 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L956](https://github.com/Minecolonies/minecolonies/blob/301f1b31147e678aaa25b31c8e5d7d826ce1c233/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L956): Reduce this anonymous class number of lines from 40 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [CitizenData.java#L23](https://github.com/Minecolonies/minecolonies/blob/301f1b31147e678aaa25b31c8e5d7d826ce1c233/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L23): class "CitizenData" has 43 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/301f1b31147e678aaa25b31c8e5d7d826ce1c233/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.264 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L46](https://github.com/Minecolonies/minecolonies/blob/301f1b31147e678aaa25b31c8e5d7d826ce1c233/src/main/java/com/minecolonies/coremod/colony/Colony.java#L46): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 44 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L46](https://github.com/Minecolonies/minecolonies/blob/301f1b31147e678aaa25b31c8e5d7d826ce1c233/src/main/java/com/minecolonies/coremod/colony/Colony.java#L46): class "Colony" has 84 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L662](https://github.com/Minecolonies/minecolonies/blob/301f1b31147e678aaa25b31c8e5d7d826ce1c233/src/main/java/com/minecolonies/coremod/colony/Colony.java#L662): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L849](https://github.com/Minecolonies/minecolonies/blob/301f1b31147e678aaa25b31c8e5d7d826ce1c233/src/main/java/com/minecolonies/coremod/colony/Colony.java#L849): Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 204 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 121 major
* ![MINOR][MINOR] 58 minor
* ![INFO][INFO] 24 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [InventoryCitizen.java#L349](https://github.com/Minecolonies/minecolonies/blob/301f1b31147e678aaa25b31c8e5d7d826ce1c233/src/main/java/com/minecolonies/coremod/inventory/InventoryCitizen.java#L349): Reduce the number of conditional operators (7) used in the expression (maximum allowed 6). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1067)
1. ![MAJOR][MAJOR] [WindowHutFarmer.java#L206](https://github.com/Minecolonies/minecolonies/blob/301f1b31147e678aaa25b31c8e5d7d826ce1c233/src/main/java/com/minecolonies/coremod/client/gui/WindowHutFarmer.java#L206): Reduce this anonymous class number of lines from 37 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L859](https://github.com/Minecolonies/minecolonies/blob/301f1b31147e678aaa25b31c8e5d7d826ce1c233/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L859): Reduce this anonymous class number of lines from 24 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L956](https://github.com/Minecolonies/minecolonies/blob/301f1b31147e678aaa25b31c8e5d7d826ce1c233/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L956): Reduce this anonymous class number of lines from 40 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [CitizenData.java#L23](https://github.com/Minecolonies/minecolonies/blob/301f1b31147e678aaa25b31c8e5d7d826ce1c233/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L23): class "CitizenData" has 43 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/301f1b31147e678aaa25b31c8e5d7d826ce1c233/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.264 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L46](https://github.com/Minecolonies/minecolonies/blob/301f1b31147e678aaa25b31c8e5d7d826ce1c233/src/main/java/com/minecolonies/coremod/colony/Colony.java#L46): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 44 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L46](https://github.com/Minecolonies/minecolonies/blob/301f1b31147e678aaa25b31c8e5d7d826ce1c233/src/main/java/com/minecolonies/coremod/colony/Colony.java#L46): class "Colony" has 84 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L662](https://github.com/Minecolonies/minecolonies/blob/301f1b31147e678aaa25b31c8e5d7d826ce1c233/src/main/java/com/minecolonies/coremod/colony/Colony.java#L662): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L849](https://github.com/Minecolonies/minecolonies/blob/301f1b31147e678aaa25b31c8e5d7d826ce1c233/src/main/java/com/minecolonies/coremod/colony/Colony.java#L849): Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Raycoms** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1154/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Comment by marchermans** [comment]
:shipit:

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1154/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Pull request merged**